enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -285,25 +285,27 @@ impl<'a> Handler<'a> {
         let buf = self.aex.gpr.rsi.raw() as *const u8;
         let size = self.aex.gpr.rdx.raw();
 
-        // Allocate some unencrypted memory.
-        let map = match self.ualloc(size) {
-            Err(errno) => return errno as u64,
-            Ok(map) => map,
-        };
-
         unsafe {
             // Copy the encrypted input into unencrypted memory.
-            copy_nonoverlapping(buf, map, size as _);
+            copy_nonoverlapping(buf, &self.block.buf as *const _ as *mut u8, size as _);
```

Comment:
Instead of converting the `buf` to a pointer, let's convert the input to a slice and use the slice copy methods. Hoorah for safe code!

Reply:
@npmccallum Do you mean something other than `let input = from_raw_parts(buf, size as usize);`? I have to also wrap that as unsafe.

Reply:
Yes, that's what I mean.